src/app/tests/TestSceneTable.cpp

```diff
@@ -0,0 +1,852 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app/clusters/scenes/SceneTableImpl.h>
+#include <lib/core/TLV.h>
+#include <lib/support/Span.h>
+#include <lib/support/TestPersistentStorageDelegate.h>
+#include <lib/support/UnitTestRegistration.h>
+#include <nlunit-test.h>
+
+using namespace chip;
+
+using SceneTable         = scenes::SceneTable;
+using SceneTableEntry    = scenes::DefaultSceneTableImpl::SceneTableEntry;
+using SceneTableImpl     = scenes::DefaultSceneTableImpl;
+using SceneStorageId     = scenes::DefaultSceneTableImpl::SceneStorageId;
+using SceneData          = scenes::DefaultSceneTableImpl::SceneData;
+using ExtensionFieldsSet = scenes::ExtensionFieldsSet;
+using TransitionTimeMs   = scenes::TransitionTimeMs;
+
+#define ON_OFF_CID 0x0006
```

Comment:
All these defines should be constexpr strongly typed expressions in the anonymous namespace and follow `kCurrentSaturationId` format instead of `CURRENT_SAT_ID`.

Reply:
Change made in latest commit.